source/fundamentals/connection/mongoclientsettings.txt

```diff
@@ -0,0 +1,201 @@
+====================
+MongoClient Behavior
+====================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+
+Overview
+--------
+
+In this guide, you can learn about the different settings to control
+the behavior of your ``MongoClient``.
+
+You can control the behavior of your ``MongoCleint`` by creating and passing in a 
+`MongoClientSettings <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.html>`__
+object to the `MongoClients.create() <{+api+}/apidocs/mongodb-driver-sync/com/mongodb/client/MongoClients.html#create(com.mongodb.MongoClientSettings)>`__
+method.
+
+To create a ``MongoClientSettings`` object, use the
+``MongoClientSettings.builder()`` method and chain methods to specify
+your desired settings. After chaining all the settings you want, use the
+``build()`` method to create the ``MongoClientSettings`` object.
+
+.. tip:: Builder Pattern
+
+   The builder pattern makes the settings immutable and supports 
+   a simpler construction through chaining.
+
+Example
+~~~~~~~
+
+This example demonstrates specifying a ``ConnectionString`` in a
+``MongoClientSettings``:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/mcs.java
+   :start-after: begin ConnectionString
+   :end-before: end ConnectionString
+   :language: java
+   :emphasize-lines: 3
+   :dedent:
+
+.. tip::
+
+   If you omit the ``applyConnectionString()`` method, the driver
+   attempts to connect to ``localhost``.
+
+For a list of methods available in ``MongoClientSettings``, 
+see the `API Documentation <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.Builder.html>`__.
+
+The following sections describe commonly used settings:
+
+- :ref:`Cluster Settings <mcs-cluster-settings>`
+- :ref:`Socket Settings <mcs-socket-settings>`
+- :ref:`Connection Pool Settings <mcs-connectionpool-settings>`
+- :ref:`Server Settings <mcs-server-settings>`
+- :ref:`SSL Settings <mcs-ssl-settings>`
+
+.. _mcs-cluster-settings:
+
+Cluster Settings
+----------------
+
+Chain the `applyToClusterSettings() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.Builder.html#applyToClusterSettings(com.mongodb.Block)>`__
+method to modify the driver's behavior when interacting with your
+MongoDB cluster.
+
+Example
+~~~~~~~
+
+This example specifies the following driver behavior in a MongoDB cluster:
+
+- To connect directly to a server, regardless of the type of cluster it is a part of
+- The primary node election to take at most ``35 SECONDS``
```

Comment:
question: Since direct connection means "connect to this server whether it is the primary or not", does it make sense to also specify a selection timeout? I think it's safe to remove the timeout and maybe mention this is analogous to directConnection and link to that.